### ISSUE TITLE
Leverage ValidityState API for validation messages

### ISSUE THREAD

**Issue by ccnokes** [react-hook-form/react-hook-form#1751]
**Is your feature request related to a problem? Please describe.**
If I have an input type field like `number`, I don't have an easy way to show a validation error related to it. 

Using this sandbox: https://codesandbox.io/s/thirsty-violet-x4pyt?file=/src/app.tsx. Run these steps:

1. Input `1..1.1` (or something invalid) in the number field
2. Submit the form
3. The ErrorMessage displays `This field is required`, presumably its defaulting to this because there is a validation error even though that's the wrong message. If you manually check the DOM's `validationMessage` property, it says `Please enter a number.`. This is correct and more helpful. 

**Describe the solution you'd like**
So the problem is to get a validation message for the number input, I have to put in a `pattern` with a regex and message. Is there a way we could leverage the DOM ValidityState API to get more accurate and helpful validation messages? (Another bonus with that API is the validation messages are localized by default). Maybe if there is no validation message set it can fallback to using the DOM one.

It looks like maybe that logic could be put in https://github.com/react-hook-form/react-hook-form/blob/master/src/errorMessage.tsx. Maybe that'd be something you explicitly opt in to via a configuration option? 

I realize this is a little vague, just thought there could be an opportunity here to make react-hook-form easier to use and leverage the platform more, so I thought I'd open a dialog about it. If we settle on something, I'm happy to PR this 😀.

**Comment by bluebill1049** [comment]
thanks @ccnokes. Thanks for the suggestion. I did actually working on native form validation (browse level) from v3.

1. during v3 the usage is low: https://react-hook-form.com/api#Browserbuiltinvalidation
2. mixed with validation with build-in and browser is confusing 
3. this will lead to inconsistent API between RN and Web. 

hence I have decided to stick with build-in validation or schema.

**solution:**

like you suggested above, either use `pattern` or `validate` function.
